### PR TITLE
Added speakers page populating through json

### LINKED ISSUE #111: Create a dedicated page for all speakers
Since all speakers are working hard to write their proposals and present them, it is worth making a page for all speakers with their image and bio and link their proposals to it. CFP team will collect the images and bio for us after proposal selection in over.

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
Fixes #111 

**Changes**
- Added `speakers.json`
- Populated html page with the mentioned json 

**testing link**
https://thick-lettuce.surge.sh/speakers.html

Template for adding speaker:
```json
    {
        "Name": "",
        "City": "",
        "Designation": "",
        "organisation": "",
        "proposal_link": "",
        "profile_image": "",
        "social_links": {
            "facebook": "",
            "twitter": "",
            "linkedin": "",
            "github": "",
            "website": ""
        }
    }
```

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
CC @ananyo2012 @vipulgupta2048 please review

**Comment by ananyo2012** [comment]
Let's just keep the Speaker image, name, designation, location in the block and show the description and links in some hover effect, since most of the times the description is bit large than expected and it becomes difficult to adjust it in the block.

**Comment by vaibhavsingh97** [comment]
@ananyo2012 Can you show me some examples 😅 as I think, we can have conditions like description should be `140` characters. Views?

**Comment by ananyo2012** [comment]
@vaibhavsingh97 Can you have a look at the keynote speakers section in the website. I was suggesting something similar to that or you can use some cool looking hover cards as well. @rajataaron @vipulgupta2048 @realslimshanky Any suggestion from your side ?

**Comment by vaibhavsingh97** [comment]
@ananyo2012 I got the point, but IMO, isn't this good as if someone wants to look at speakers than he is able to see without hovering on the card itself. Also, We can play with colors and font-size so that Description is more clear to see.  As adding information as shown in keynote speakers section is sometimes difficult to read due to color combination. 
@ananyo2012 @vipulgupta2048 @realslimshanky @rajataaron View please

**Comment by ananyo2012** [comment]
Yeah I am just concerned that people have to scroll a lot more if the number is large. This looks good IMO, probably you can make it bit more compact.

**Comment by vaibhavsingh97** [comment]
@ananyo2012 This will result in inner content scrollable and this would not look gud

**Comment by vipulgupta2048** [comment]
I apologize for not commenting sooner. Here are my thoughts.
@ananyo2012 How many speakers do you think will be there, if we look at the previous PyCon stats there were a total of 57 speakers, keynotes included. 

> Yeah I am just concerned that people have to scroll a lot more if the number is large. This looks good IMO, probably you can make it bit more compact.

Hence correct, of what you propose. But the counter-argument of the inner content being scrollable won't look good either and is neither feasible. So here's what I propose. 
1. Drop the description like @vaibhavsingh97 mentioned. I feel having description of so many speakers will be technically a JSON Nightmare, unless the JSON updates itself from Junction itself. 
2. Keep the description, but store it in a hover box like Keynote speakers. The page will become too long and not very easy to scroll. If you look at mobile version, then having 50+ speakers in a stack is a UI nightmare. 

- Also, there is an issue of having **limited description**. We can't have a complete description of everyone. Hence we can either remove it completely or keep it but it would be incomplete. Personally, the links below would describe the person better. 
- I feel we should also a give a **link to their talk** because personally, I would like to know what he would be speaking about more than to know him/her. This is the PyCon way of doing things, check it out here https://us.pycon.org/2018/schedule/talks/list/

**Comment by ananyo2012** [comment]
@vaibhavsingh97 Yes I agree with @vipulgupta2048 's points. Let's remove description from the main block and show it on hover. Also add the link to the proposal instead.

**Comment by realslimshanky** [comment]
@vaibhavsingh97 any update on the suggestions given by @vipulgupta2048? Also you need to rebase your branch.

**Comment by vaibhavsingh97** [comment]
@realslimshanky Yeah, I totally forgot, Will update this PR. But I am still confused how I show description in hover. Also, What's the point of showing description if we are giving the link of proposals. I mean we can just give the link of proposal and user can navigate to the speaker descriptio

**Comment by ananyo2012** [comment]
Yes let's just drop this description thing. Just link them to the proposal.
Rest all is fine.

**Comment by vaibhavsingh97** [comment]
@vipulgupta2048 @ananyo2012 @realslimshanky Updated the codebase according to the changes requested. Please review 😄

**Comment by vaibhavsingh97** [comment]
@realslimshanky What If new contributors should do this? We can open one master issue and will add the link of the proposal which we want to add as a speaker. This will benefit both the project and contributor

**Comment by ananyo2012** [comment]
@vaibhavsingh97 This looks good! Just remove the facebook button. People are moving away from fb now 😛 . @realslimshanky I think we need to modify the navbar now that so many pages have come up. Also for now can we add the details of Keynote speakers in speakers.json ?

**Comment by vaibhavsingh97** [comment]
@ananyo2012 I haven't added, if you don't add the Facebook profile in json than it will not appear on the page.

**Comment by realslimshanky** [comment]
@vaibhavsingh97 whichever suits the project best.
@ananyo2012 yeah. We need to reassign the URL of nav. Also, we can add keynote in the `/data/speakers.json and display 2 separate sections inside `/speakers.html` one for keynote and other for rest of the speakers. And since rest of the speakers are yet to be announced we can add a placeholder to the second section.

**Comment by vaibhavsingh97** [comment]
@realslimshanky Cool, I am thinking of the same. I will add the `keynote` section and a placeholder section. Also, Should I remove the `keynote` section in `index` page?

**Comment by realslimshanky** [comment]
@vaibhavsingh97 no. Don't remove Keynote from the index. It's important that it stays there as well.

File: js/speaker.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+$(document).ready(function() {
+  fetch('../data/speakers.json')
+    .then(response => response.json())
```
Comment (ananyo2012) [inline js/speaker.js]: Please do not use arrow functions. We are not compiling the javascript code. Just use `function` instead.

File: js/speaker.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            '" alt="profile-sample2" /></div><figcaption><h3>' +
+            item.Name +
+            '</h3><h5>' +
+            item.Designation +
+            '</h5><span><i class="fas fa-map-marked-alt"></i>' +
+            item.City +
+            '</span><p id="proposal-link"><i class="fas fa-file-alt"></i> <a href="' +
+            item.proposal_link +
+            '">Speaker Bio</a></p><div class="icons">';
+          if (item.social_links.facebook) {
```
Comment (ananyo2012) [inline js/speaker.js]: Remove the facebook icons and code.

**Comment by ananyo2012** [comment]
@realslimshanky Let's not create different sections. Just keep the keynotes on top. @vaibhavsingh97 Can you add the Keynote speakers in the json file ?

**Comment by realslimshanky** [comment]
@ananyo2012 by creating sections I mean giving just a header 'Keynote' and 'Speakers' etc.

**Comment by vaibhavsingh97** [comment]
Yeah, I will send PR adding suggestions 😀

**Comment by realslimshanky** [comment]
@vaibhavsingh97 please update about your progress and resolve conflicts while you are working with the PR.

**Comment by vaibhavsingh97** [comment]
@realslimshanky Yes, I will update, sorry for keeping this PR so long.

File: js/speaker.js (hunk 2 above)
Comment (vaibhavsingh97) [reply js/speaker.js]: I will remove it

File: js/speaker.js (hunk 1 above)
Comment (vaibhavsingh97) [reply js/speaker.js]: @ananyo2012 It's the ES2016 standard, and it should be followed 😅. But if you want I will remove

**Comment by vaibhavsingh97** [comment]
@realslimshanky @ananyo2012 What should go in place of the `proposal` in keynote speakers?

File: js/speaker.js (hunk 1 above)
Comment (ananyo2012) [reply js/speaker.js]: Yes if you are compiling the code then you should follow ES6. But we are not doing any compilation here. So you have to use the basic syntax as ES6 is still not supported in many browsers.

**Comment by ananyo2012** [comment]
@vaibhavsingh97 Lets make the json something like this. This way we can categorize everything as well
```json
{
   "Keynote" : {
      "Name": "",
      "City": "",
      ...
   },
   "Workshops" : {
      "Name": "",
      "City": "",
      ...
   },
   "Tracks" : {
      "Name": "",
      "City": "",
      ...
   }
}
```

**Comment by ananyo2012** [comment]
@vaibhavsingh97 Any update on this ?

**Comment by vaibhavsingh97** [comment]
Hi! Sorry for keeping so long, I will update by tomorrow for sure. You can
expect update by evening 😅

**Comment by vaibhavsingh97** [comment]
@ananyo2012 @realslimshanky Please review, I have added suggested changes.
Review link: https://stupendous-education.surge.sh/speakers.html

**Comment by vaibhavsingh97** [comment]
JSON Structure
```json
{
    "keynote speakers": [
        {
            "Name": "",
            "City": "",
            "Designation": "",
            "profile_image": "",
            "Description": "",
            "social_links": {
                "twitter": "https://twitter.com/",
                "linkedin": "https://www.linkedin.com/in/",
                "github": "http://github.com/",
                "website": ""
            }
        }
    ],
    "speakers": [
        {
            "Name": "",
            "City": "",
            "Designation": "",
            "proposal_link": "",
            "profile_image": "",
            "social_links": {
                "twitter": "https://twitter.com",
                "linkedin": "https://linkedin.com",
                "github": "https://github.com",
                "website": "https://test.com"
            }
        }
    ]
}
```

**Comment by ananyo2012** [comment]
Merging this in. @vaibhavsingh97 Thanks for your contribution on this!

**Pull request merged**